flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecSinkITCase.java

```diff
@@ -187,6 +198,148 @@ public void testStreamRecordTimestampInserterNotApplied() {
         assertPlan(tableEnv, "INSERT INTO T1 SELECT * FROM T1", false);
     }
 
+    @Test
+    public void testCharPrecisionEnforcer() throws ExecutionException, InterruptedException {
+        final StreamTableEnvironment tableEnv = StreamTableEnvironment.create(env);
+        final List<Row> rows =
+                Arrays.asList(
+                        Row.of(1, "Apache Flink", "SQL RuleZ", 11, 111, "SQL"),
+                        Row.of(2, "Apache", "SQL", 22, 222, "Flink"),
+                        Row.of(3, "Apache", "Flink SQL", 33, 333, "Apache Flink SQL"),
+                        Row.of(4, "Flink Project", "SQL or SeQueL?", 44, 444, "Apache Flink SQL"));
+
+        final TableDescriptor sourceDescriptor =
+                TableFactoryHarness.newBuilder()
+                        .schema(schemaForCharPrecisionEnforcer())
+                        .source(new TestSource(rows))
+                        .build();
+        tableEnv.createTable("T1", sourceDescriptor);
+
+        // Default config - ignore (no trim)
+        TableResult result = tableEnv.executeSql("SELECT * FROM T1");
+        result.await();
+
+        final List<String> expected = rows.stream().map(Row::toString).collect(Collectors.toList());
+        final List<String> resultsAsString = new ArrayList<>();
+        result.collect().forEachRemaining(r -> resultsAsString.add(r.toString()));
+        assertEquals(expected, resultsAsString);
```

Comment:
We should compare this regardless of order for stability. Also, I wonder if we can't extract this entire block into a reusable piece?

Reply:
I used a `containsInAnyOrder` on the Row directly instead of using the string representation. I think it's better to wait for `AssertJ` coming soon, to use a nice Row comparison method from a utility.